src/main/java/net/sf/jabref/gui/desktop/os/DefaultDesktop.java

```diff
@@ -3,31 +3,37 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
+import net.sf.jabref.external.ExternalFileType;
+
 import java.awt.*;
 import java.io.File;
 import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+public class DefaultDesktop extends NativeDesktop {
 
-public class DefaultDesktop implements NativeDesktop {
     private static final Log LOGGER = LogFactory.getLog(NativeDesktop.class);
 
+
     @Override
-    public void openFile(String filePath, String fileType) throws IOException {
-        Desktop.getDesktop().open(new File(filePath));
+    public void openFile(String filePath, Optional<ExternalFileType> fileType) throws IOException {
```

Comment:
Never pass an optional. Create two methods instead, one with and one without a file type.